cpp/oneapi/dal/algo/pca/backend/cpu/train_kernel_cov.cpp

```diff
@@ -37,76 +37,118 @@ namespace daal_pca = daal::algorithms::pca;
+    const auto daal_singular_values =
+        interop::convert_to_daal_homogen_table(arr_singular_values, 1, component_count);
+    const auto daal_explained_variances_ratio =
+        interop::convert_to_daal_homogen_table(arr_explained_variances_ratio, 1, component_count);
 
     daal_cov::Batch<Float, daal_cov::defaultDense> covariance_alg;
-    covariance_alg.input.set(daal_cov::data, daal_data);
-
-    daal_cov::internal::Hyperparameter daal_hyperparameter;
-    /// the logic of block size calculation is copied from DAAL,
-    /// to be changed to passing the values from the performance model
-    std::int64_t blockSize = 140;
-    if (ctx.get_enabled_cpu_extensions() == dal::detail::cpu_extension::avx512) {
-        const std::int64_t row_count = data.get_row_count();
-        if (5000 < row_count && row_count <= 50000) {
-            blockSize = 1024;
-        }
-    }
-    interop::status_to_exception(
-        daal_hyperparameter.set(daal_cov::internal::denseUpdateStepBlockSize, blockSize));
-    covariance_alg.setHyperparameter(&daal_hyperparameter);
 
-    constexpr bool is_correlation = false;
-    constexpr std::uint64_t results_to_compute =
-        std::uint64_t(daal_pca::mean | daal_pca::variance | daal_pca::eigenvalue);
+    covariance_alg.parameter.outputMatrixType = daal::algorithms::covariance::correlationMatrix;
+
+    daal::algorithms::pca::BaseBatchParameter daal_pca_parameter;
+
+    daal_pca_parameter.isDeterministic = desc.get_deterministic();
 
+    daal_pca_parameter.resultsToCompute = static_cast<DAAL_UINT64>(
+        std::uint64_t(daal_pca::mean | daal_pca::variance | daal_pca::eigenvalue));
+
+    daal_pca_parameter.isCorrelation = true;
+
+    if (desc.get_normalization_mode() == normalization::mean_center) {
+        daal_pca_parameter.doScale = false;
+        daal_parameter.outputMatrixType = daal_cov::covarianceMatrix;
+    }
+    auto arr_cov_matrix = array<Float>::empty(column_count * column_count);
+    const auto daal_cov_matrix =
+        interop::convert_to_daal_homogen_table(arr_cov_matrix, column_count, column_count);
+    interop::status_to_exception(
+        interop::call_daal_kernel<Float, daal_cov_kernel_t>(ctx,
+                                                            daal_data.get(),
+                                                            daal_cov_matrix.get(),
+                                                            daal_means.get(),
+                                                            &daal_parameter));
+    {
+        const auto status = dal::backend::dispatch_by_cpu(ctx, [&](auto cpu) {
+            constexpr auto cpu_type = interop::to_daal_cpu_type<decltype(cpu)>::value;
+            return daal_pca_cor_kernel_t<Float, cpu_type>().computeVariancesFromCov(
+                *daal_cov_matrix,
+                *daal_variances);
+        });
+        interop::status_to_exception(status);
+    }
```

Comment:
This code snippet is duplicated in different files. Consider using a separate `.hpp` file to isolate and reuse such things